vivisect/analysis/__init__.py

```diff
@@ -82,7 +83,10 @@ def addAnalysisModules(vw):
             viv_analysis_i386.addEntrySigs(vw)
             vw.addAnalysisModule("vivisect.analysis.i386.importcalls")
             # add va set for tracking thunk_bx function(s)
-            vw.addVaSet('thunk_bx', ( ('fva', vivisect.VASET_ADDRESS), ) )
+            try:
+                vw.getVaSet('thunk_bx')
+            except:
+                vw.addVaSet('thunk_bx', ( ('fva', vivisect.VASET_ADDRESS), ) )
```

Comment:
totally agreed

Reply:
on second thought...

so, this is really only pertinent for i386.
unfortunately, we don't currently know the architecture when we call VivWorkspace.__init__()
so... this seems to be the best place to add VaSets based on arch/platform.

Reply:
that's great in concept.  currently these are created in the VivWorkspace.__init__() before Architecture and Platform are set.  
if you'd like, we can move them to an analysis modules called later on, but before any real analysis happens (so that they are all initialized when we started writing to them)?